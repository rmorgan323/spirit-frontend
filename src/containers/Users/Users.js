import React from 'react';
import { connect } from 'react-redux';
import { PropTypes } from 'prop-types';
import './Users.css';

export const Users = props => {
  if (props.user.id) {
    const userId = props.user.id;
    props.history.push(`/spirit/users/${userId}/`);
  }

  return <div />;
};

export const mapStateToProps = store => ({
  user: store.user
});

export default connect(mapStateToProps, null)(Users);

Users.propTypes = {
  user: PropTypes.object,
  history: PropTypes.object
};
