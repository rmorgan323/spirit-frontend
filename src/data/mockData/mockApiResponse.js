const mockApiResponse = {
  status: 'success'
};

export default mockApiResponse;