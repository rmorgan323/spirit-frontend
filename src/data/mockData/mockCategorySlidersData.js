/*eslint-disable camelcase*/

export const mockTitle = 'Capacities';

export const mockSlidersArray = [
  'Attention',
  'Action',
  'Affect',
  'Autonomic',
  'Arousability'
];

export const mockDatabaseNamesArray = [
  'mod_2_attention',
  'mod_2_action',
  'mod_2_affect',
  'mod_2_autonomic',
  'mod_2_arousability'
];

export const mockReminderAsterisk = [null, '*', '*', '*', '*'];

export const mockReminder = '* If selected, remember to make a hypothesized arousal chart.';

export const mockSliderTitle = 'Attention';

export const mockDatabaseName = 'mod_2_attention';

