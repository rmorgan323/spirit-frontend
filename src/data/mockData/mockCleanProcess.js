/*eslint-disable camelcase*/

export const mockCleanProcess = {
  id: 10,
  sen_h_vestibular: null,
  sen_h_proprioception: null,
  sen_h_tactile: null,
  sen_h_auditory: null,
  sen_h_visual: null,
  sen_h_intero: null,
  sen_1_ability: null,
  sen_1_match: null,
  sen_1_classification: null,
  sen_1_spatial: null,
  sen_1_figure: null,
  sen_1_inferences: null,
  sen_2_initiation: null,
  sen_2_imitation: null,
  sen_2_affordance: null,
  sen_2_generativity: null,
  sen_3_sequencing: null,
  sen_3_sequencing_gestalt: null,
  sen_3_schema_selection: null,
  sen_3_schema_expansion: null,
  sen_3_modification: null,
  sen_4_timing: null,
  sen_4_timing_force: null,
  sen_4_spatial: null,
  sen_4_temporal: null,
  sen_4_control: null,
  sen_4_monitoring: null,
  sen_5_oral: null,
  sen_5_ocular: null,
  sen_5_postural: null,
  sen_5_limb: null,
  sen_5_gross: null,
  sen_5_fine: null,
  sen_5_tool: null,
  sen_5_projected: null,
  sen_6_somatodyspraxia: null,
  sen_6_visual: null,
  sen_6_generalized: null,
  sen_6_bmc: null,
  sen_6_praxis: null,
  mod_h_vestibular: null,
  mod_h_proprioception: null,
  mod_h_tactile: null,
  mod_h_auditory: null,
  mod_h_visual: null,
  mod_h_intero: null,
  mod_1_defensiveness: null,
  mod_1_over: null,
  mod_1_under: null,
  mod_1_variable: null,
  mod_2_attention: null,
  mod_2_action: null,
  mod_2_affect: null,
  mod_2_autonomic: null,
  mod_3_physiological: null,
  mod_3_sensory: null,
  mod_3_motor: null,
  mod_3_emotional: null,
  mod_3_cognitive: null,
  mod_3_social: null,
  mod_3_prosocial: null,
  mod_4a_passive: null,
  mod_4a_co_regulation: null,
  mod_4a_approach: null,
  mod_4a_adaptive: null,
  mod_4b_active: null,
  mod_4b_self_regulation: null,
  mod_4b_avoid: null,
  mod_4b_maladaptive: null,
  exe_1_orient: null,
  exe_1_inhibition: null,
  exe_2a_working: null,
  exe_2a_shifting: null,
  exe_2b_verbal: null,
  exe_2b_spatial: null,
  exe_2b_general: null,
  exe_2b_global: null,
  exe_2b_flexibility: null,
  exe_3_sequencing: null,
  exe_3_monitoring: null,
  exe_4a_future: null,
  exe_4a_spatial: null,
  exe_4a_problem: null,
  exe_4a_reflective: null,
  exe_4a_internalized: null,
  exe_4b_motivational: null,
  exe_4b_effortful: null,
  exe_4b_self_control: null,
  exe_4b_self_regulation: null,
  exe_4b_self_compassion: null,
  pos_h_vestibular: null,
  pos_h_proprioception: null,
  pos_h_tactile: null,
  pos_h_auditory: null,
  pos_h_visual: null,
  pos_h_intero: null,
  pos_2_stability: null,
  pos_2_mobility: null,
  pos_2_tonic: null,
  pos_2_phasic: null,
  pos_2_muscle: null,
  pos_3_sagittal: null,
  pos_3_frontal: null,
  pos_3_transverse: null,
  pos_3_midlines: null,
  pos_5_suck: null,
  pos_5_ocular: null,
  pos_5_antigravity: null,
  pos_5_reflex: null,
  pos_5_alignment_COG: null,
  pos_5_alignment_orient: null,
  pos_5_anticipatory: null,
  pos_5_interactive: null,
  pos_6_push: null,
  pos_6_bilateral: null,
  pos_6_grasp: null,
  pos_6_strength: null,
  pos_6_effortful: null,
  soc_1_regulation: null,
  soc_1_engagement: null,
  soc_1_reciprocity: null,
  soc_1_problem: null,
  soc_1_representation: null,
  soc_1_triangular: null,
  soc_1_comparative: null,
  soc_1_reflective: null,
  soc_2_self_awareness: null,
  soc_2_joint: null,
  soc_2_social_referencing: null,
  soc_2_imitation: null,
  soc_2_inferential: null,
  soc_2_contingent: null,
  soc_2_theory: null,
  soc_2_central: null,
  soc_2_mentalizing: null,
  soc_2_empathy: null,
  soc_2_social_motivators: null,
  soc_2_pro_social: null,
  soc_3_affective: null,
  soc_3_emotional_contagion: null,
  soc_3_activation_low: null,
  soc_3_emotional_cascade: null,
  soc_3_activation_high: null,
  soc_3_emotional_differentiation: null,
  soc_3_emotional_association: null,
  soc_3_self_control: null,
  soc_3_self_regulation: null,
  soc_3_self_compassion: null,
  soc_4_mental: null,
  soc_4_thoughts: null,
  soc_4_attachments: null,
  soc_4_trauma: null,
  session_id: 10,
  created_at: "2018-02-20T21:25:45.454Z",
  updated_at: "2018-02-20T21:25:45.454Z",
  mod_2_arousability: null,
  exe_3_initiation: null,
  exe_3_planning: null,
  pos_4_core: null,
  pos_4_shoulder: null,
  pos_4_pelvic: null,
  pos_4_head: null,
  pos_4_eyes: null,
  pos_4_hand: null,
  pos_4_lower: null,
  pos_4_foot: null,
  soc_2_social_perspective: null,
  pos_1_outer: null,
  pos_1_inner: null,
  exe_4a_generativity: null
};
