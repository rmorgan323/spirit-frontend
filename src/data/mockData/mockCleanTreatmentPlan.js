/*eslint-disable camelcase*/

export const mockCleanTreatmentPlan = {
  id: 10,
  modulation_sensory: null,
  modulation_task: null,
  modulation_environment: null,
  modulation_predictability: null,
  modulation_self_regulation: null,
  modulation_interaction: null,
  modulation_JRC_AR_notes: null,
  posture_sensory: null,
  posture_task: null,
  posture_environment: null,
  posture_predictability: null,
  posture_self_regulation: null,
  posture_interaction: null,
  posture_JRC_AR_notes: null,
  sensory_discrimination_sensory: null,
  sensory_discrimination_task: null,
  sensory_discrimination_environment: null,
  sensory_discrimination_predictability: null,
  sensory_discrimination_self_regulation: null,
  sensory_discrimination_interaction: null,
  sensory_discrimination_JRC_AR_notes: null,
  social_emotional_sensory: null,
  social_emotional_task: null,
  social_emotional_environment: null,
  social_emotional_predictability: null,
  social_emotional_self_regulation: null,
  social_emotional_interaction: null,
  social_emotional_JRC_AR_notes: null,
  executive_functioning_sensory: null,
  executive_functioning_task: null,
  executive_functioning_environment: null,
  executive_functioning_predictability: null,
  executive_functioning_self_regulation: null,
  executive_functioning_interaction: null,
  executive_functioning_JRC_AR_notes: null,
  session_id: 10,
  created_at: "2018-02-20T21:25:45.468Z",
  updated_at: "2018-02-20T21:25:45.468Z"
};
