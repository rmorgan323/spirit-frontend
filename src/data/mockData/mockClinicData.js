/*eslint-disable camelcase*/

export const mockClinicData = {
  name: 'I Need Therapy',
  abbreviation: 'INT',
  clinic_id: 10,
  passcode: 'f83jds8e'
};
