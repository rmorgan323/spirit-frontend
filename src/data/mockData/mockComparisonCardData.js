export const mockDataArray = [
  "2018-02-20T20:08:23.713Z",
  "2018-02-20T20:12:14.588Z",
  "2018-02-20T21:25:45.454Z"
];
export const mockTitle = "Maladaptive";
export const mockValues = ["5r", "6i", "5f"];