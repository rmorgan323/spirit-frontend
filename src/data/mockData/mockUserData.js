/*eslint-disable camelcase*/

export const mockUserData = {
  id: 1,
  authrocket_id: "usr_0vBdVVz3RUsbR2ArDCoPvo",
  name: "Lee Trevino",
  email: "sample@samplemail.com",
  admin: false,
  created_at: "2018-02-19T03:41:23.701Z",
  updated_at: "2018-02-19T03:41:23.701Z",
  clinic: "I Need Therapy",
  clinic_abbreviation: "INT",
  clinic_id: 10,
  clinic_passcode: "utk85fmf"
};
