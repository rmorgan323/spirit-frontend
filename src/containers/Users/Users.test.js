import React from 'react';
import { Users, mapStateToProps } from './Users';
import { shallow } from 'enzyme';
import { mockUserData } from '../../data/mockData/mockUserData';
import { mockHistory } from '../../data/mockData/mockHistory';

describe('Users tests', () => {
  let user;
  let history;
  let renderedUsers;

  beforeEach(() => {
    user = mockUserData;
    history = mockHistory;

    renderedUsers = shallow(<Users user={user} history={history} />);
  });

  it('Should match the snapshot', () => {
    expect(renderedUsers).toMatchSnapshot();
  });

  it('Should redirect if there is a user', () => {
    expect(history.push).toHaveBeenCalled();
  });
});

describe('mapStateToProps tests', () => {
  it('Should pull a user from store', () => {
    const user = mockUserData;
    const mockStore = { user };
    const result = mapStateToProps(mockStore);

    expect(result.user).toEqual(mockStore.user);
  });
});
