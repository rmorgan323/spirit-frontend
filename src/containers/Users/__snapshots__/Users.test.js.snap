// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Users tests Should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    history={
      Object {
        "push": [Function],
      }
    }
    user={
      Object {
        "admin": false,
        "authrocket_id": "usr_0vBdVVz3RUsbR2ArDCoPvo",
        "clinic": "I Need Therapy",
        "clinic_abbreviation": "INT",
        "clinic_id": 10,
        "clinic_passcode": "utk85fmf",
        "created_at": "2018-02-19T03:41:23.701Z",
        "email": "sample@samplemail.com",
        "id": 1,
        "name": "Lee Trevino",
        "updated_at": "2018-02-19T03:41:23.701Z",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {},
    "ref": null,
    "rendered": null,
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {},
      "ref": null,
      "rendered": null,
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
